### DISCUSSION TITLE
Integration API Hook 'astro:server:update'

### DISCUSSION THREAD

**Discussion by vinnichase** [withastro/roadmap#738] [Proposal]
# Summary

This proposal suggests a new hook for the integration API 'astro:server:update'. 

# Background & Motivation

For integrations that use the source file tree to generate structures during build it would be helpful to listen on update events to enable hot reloading for the developer while he is editing source files. This could for example also help the sitemap integration.

# Goals

Every time a source file is updated the above mentioned hook should be fired along with the file path that is also shown in the console (`11:40:31 AM [astro] update /src/pages/index.astro`)

**Comment by florian-lefebvre** [comment]
Isn't `server.watcher.on` enough through https://docs.astro.build/en/reference/integrations-reference/#server-option ?

**Reply by vinnichase** [reply]
Ok indeed, thats exactly what I was looking for. thanks for the quick reply!
